File: twoSum/two_sum.go
func twoSum(nums []int, target int) []int {
	d := make(map[int][2]int)
	for i, v := range nums {
			value, ok := d[v]
			if (ok && value[0] * 2 == target) {
					return []int{value[1], i}
			}
			d[v] = [2]int{target - v, i}
	}
	
	for _, v := range d {
			_, ok := d[v[0]]

			if ok {
					complement_index := d[v[0]][1]
					if complement_index != v[1] {
							return []int{v[1], complement_index}
					}
			}
	}

	return []int{0, 0}
}

File: twoSum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, num in enumerate(nums):
            value = d.get(num)
            if value is not None:
                if value[0] * 2 == target:
                    return [value[1], i]

            d[num] = (target - num, i)

        for k, v in d.items():
            complement, index = v
            if complement in d:
                complement_index = d[complement][1]
                if complement_index != index: 
                    return [index, complement_index]
